internal/databases/mongo/delete_handler.go

```diff
@@ -105,7 +105,12 @@ func HandleBackupsPurge(backupTimes []internal.BackupTime,
 		return nil, nil, err
 	}
 
-	purge, retain, err = archive.SplitPurgingBackups(backups, opts.retainCount, opts.retainAfter)
+	timedBackups := archive.MongoModelToTimedBackup(backups)
+
+	purgeBackups, retainBackups, err := internal.SplitPurgingBackups(timedBackups, opts.retainCount, opts.retainAfter)
+	purge = archive.TimedBackupToMongoModel(purgeBackups)
```

Comment:
Let's try to avoid assertions in the following functions.
There are two []Timed arrays. Now we can build two sets (map[backup_name]struct{}) from them and check if each item from []MongoModelBackup should be deleted or retained.